go/ql/test/library-tests/semmle/go/frameworks/Fasthttp/fasthttp.go

```diff
@@ -0,0 +1,218 @@
+		// requestCtx.MultipartForm()
+		requestCtx.URI().Path()            // $ UntrustedFlowSource="call to Path"
+		requestCtx.URI().PathOriginal()    // $ UntrustedFlowSource="call to PathOriginal"
+		requestCtx.URI().FullURI()         // $ UntrustedFlowSource="call to FullURI"
+		requestCtx.URI().LastPathSegment() // $ UntrustedFlowSource="call to LastPathSegment"
+		requestCtx.URI().QueryString()     // $ UntrustedFlowSource="call to QueryString"
+		requestCtx.URI().String()          // $ UntrustedFlowSource="call to String"
+
+		//or requestCtx.PostArgs()
+		requestCtx.URI().QueryArgs().Peek("arg1")                   // $ UntrustedFlowSource="call to Peek"
+		requestCtx.URI().QueryArgs().PeekBytes([]byte("arg1"))      // $ UntrustedFlowSource="call to PeekBytes"
+		requestCtx.URI().QueryArgs().PeekMulti("arg1")              // $ UntrustedFlowSource="call to PeekMulti"
+		requestCtx.URI().QueryArgs().PeekMultiBytes([]byte("arg1")) // $ UntrustedFlowSource="call to PeekMultiBytes"
+		requestCtx.URI().QueryArgs().QueryString()                  // $ UntrustedFlowSource="call to QueryString"
+		requestCtx.URI().QueryArgs().String()                       // $ UntrustedFlowSource="call to String"
+		requestCtx.String()                                         // $ UntrustedFlowSource="call to String"
+
+		requestCtx.Path() // $ UntrustedFlowSource="call to Path"
+		// multipart.Form is already implemented
+		// requestCtx.FormFile("FileName")
+		// requestCtx.FormValue("ValueName")
+		requestCtx.Referer()           // $ UntrustedFlowSource="call to Referer"
+		requestCtx.PostBody()          // $ UntrustedFlowSource="call to PostBody"
+		requestCtx.RequestBodyStream() // $ UntrustedFlowSource="call to RequestBodyStream"
+		requestCtx.RequestURI()        // $ UntrustedFlowSource="call to RequestURI"
+		requestCtx.UserAgent()         // $ UntrustedFlowSource="call to UserAgent"
+		requestCtx.Host()              // $ UntrustedFlowSource="call to Host"
+
+		requestCtx.Request.Host()                         // $ UntrustedFlowSource="call to Host"
+		requestCtx.Request.Body()                         // $ UntrustedFlowSource="call to Body"
+		requestCtx.Request.RequestURI()                   // $ UntrustedFlowSource="call to RequestURI"
+		body1, _ := requestCtx.Request.BodyGunzip()       // $ UntrustedFlowSource="... := ...[0]"
+		body2, _ := requestCtx.Request.BodyInflate()      // $ UntrustedFlowSource="... := ...[0]"
+		body3, _ := requestCtx.Request.BodyUnbrotli()     // $ UntrustedFlowSource="... := ...[0]"
+		body4, _ := requestCtx.Request.BodyUncompressed() // $ UntrustedFlowSource="... := ...[0]"
+		requestCtx.Request.BodyStream()                   // $ UntrustedFlowSource="call to BodyStream"
+		requestCtx.Request.ReadBody(dstReader, 100, 1000)
+		requestCtx.Request.ReadLimitBody(dstReader, 100)
+		requestCtx.Request.ContinueReadBodyStream(dstReader, 100, true)
+		requestCtx.Request.ContinueReadBody(dstReader, 100)
```

Comment:
Yes, I recommend it. Tests are quite important for CodeQL because it's so easy to write something that works in your head, and make one small mistake and it actually does nothing. Typos in method names are an obvious example.